hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/BasicAuthConfigurator.java

```diff
@@ -33,6 +33,16 @@ public class BasicAuthConfigurator implements ConnectionConfigurator {
   private final ConnectionConfigurator parent;
   private final String credentials;
 
+  static public ConnectionConfigurator getConfigurator(
```

Comment:
Since this method has to be called by URLConnectionFactory which is in same package. Lets have it package-protected and remove `public`

Reply:
Should this method be moved to `URLConnectionFactory` class only?
Reason being, when external class calls ` BasicAuthConfigurator.getConfigurator`, to developer it can look that we want something out of `BasicAuthConfigurator`, but instead it either gives BasicAuthConfigurator object or parent-configurator object. What you feel?

Reply:
Good point, we might as well move it to `URLConnectionFactory` and make it private there. I've made the change.